src/lib.rs

```diff
@@ -210,16 +303,35 @@ impl Contract {
         res
     }
 
+    pub fn get_proposals_by_author(&self, author: AccountId) -> Vec<ProposalId> {
+        near_sdk::log!("get_proposals_by_author");
+        self.author_proposals.get(&author).map(|proposals| proposals.into_iter().collect()).unwrap_or(Vec::new())
+    }
+
+    pub fn get_proposals_by_label(&self, label: String) -> Vec<ProposalId> {
+        near_sdk::log!("get_proposals_by_label");
+        let mut res: Vec<_> =
+            self.label_to_proposals.get(&label).unwrap_or_default().into_iter().collect();
+        res.sort();
+        res
+    }
+
     pub fn get_all_labels(&self) -> Vec<String> {
         near_sdk::log!("get_all_labels");
-        let mut res: Vec<_> = self.label_to_posts.keys().collect();
+        let mut post_labels: HashSet<_> = self.label_to_posts.keys().collect::<HashSet<_>>();
+        let proposals_labels: HashSet<_> = self.label_to_proposals.keys().collect::<HashSet<_>>();
+        post_labels.extend(proposals_labels);
+        let mut res: Vec<_> = post_labels.into_iter().collect();
         res.sort();
         res
     }
 
     pub fn get_all_authors(&self) -> Vec<AccountId> {
         near_sdk::log!("get_all_authors");
-        let mut res: Vec<_> = self.authors.keys().collect();
+        let mut author_posts: HashSet<_> = self.authors.keys().collect::<HashSet<_>>();
+        let author_proposals: HashSet<_> = self.author_proposals.keys().collect::<HashSet<_>>();
+        author_posts.extend(author_proposals);
+        let mut res: Vec<_> = author_posts.into_iter().collect();
         res.sort();
         res
     }
```

Comment:
I suggest we don't touch these and the methods below, instead, create a separate set of methods for proposals.